### PR TITLE
Defect Fix for optional value

### PR DESCRIPTION
**Pull request by AmanPrasad43**
Thank you for your contribution.
Before submitting this PR, please make sure:

- [ ] PR description and commit message should describe the changes done in this PR
- [ ] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [ ] Latest Code from master or specific release branch is merged to your branch
- [ ] No unwanted\commented\junk code is included
- [ ] No new warning upon build solution
- [ ] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve all Codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Update the Help Library document to match any feature changes

## Summary by CodeRabbit

- **Refactor**
	- Enhanced API handling capabilities, including improved example value processing and enumeration support.

### CODE REVIEW

File: Ginger/GingerAutoPilot/APIModelLib/SwaggerApi/OpenApiBase.cs [hunk 1]
```diff
@@ -73,6 +75,7 @@ public void GenerateResponse(SwaggerOperation operation, ApplicationAPIModel bas
                             arv.DoNotConsiderAsTemp = true;
                             basicModal.ReturnValues.Add(arv);
                         }
+                        var test = operation.RequestBody.Content;
```
Comment (Maheshkale447) [inline Ginger/GingerAutoPilot/APIModelLib/SwaggerApi/OpenApiBase.cs:78]: Please remove this test variable if not being used

File: Ginger/GingerAutoPilot/APIModelLib/SwaggerApi/OpenApiBase.cs [hunk 2]
```diff
@@ -281,23 +285,58 @@ public ObservableList<OptionalValue> GetListOfParamEnums(SwaggerParameter swagge
...
-                if (swaggerParameter.Enumeration != null && swaggerParameter.Enumeration.Count > 0)
+                else if (swaggerParameter.Enumeration != null && swaggerParameter.Enumeration.Count > 0)
                 {
                     foreach (object item in swaggerParameter.Enumeration)
                     {
                         OptionalValue value = new OptionalValue()
                         {
                             Value = item.ToString(),
                             ItemName = item.ToString(),
+                            IsDefault = true
```
Comment (Maheshkale447) [inline Ginger/GingerAutoPilot/APIModelLib/SwaggerApi/OpenApiBase.cs:301]: Please check if its setting all values to default as true

File: Ginger/GingerAutoPilot/APIModelLib/SwaggerApi/OpenApiBase.cs [hunk 3]
```diff
@@ -281,23 +285,58 @@ public ObservableList<OptionalValue> GetListOfParamEnums(SwaggerParameter swagge
...
                         };
                         lstOptions.Add(value);
                     }
 
                 }
+                else if (swaggerParameter.ActualSchema.ActualProperties.Count>0 || swaggerParameter.ActualSchema.Enumeration.Count>0)
```
Comment (Maheshkale447) [inline Ginger/GingerAutoPilot/APIModelLib/SwaggerApi/OpenApiBase.cs:307]: Same condition is being checked in below if blocks, please verify if two conditions are needed

File: Ginger/GingerAutoPilot/APIModelLib/SwaggerApi/OpenApiBase.cs [hunk 4]
```diff
@@ -306,13 +345,50 @@ public ObservableList<OptionalValue> GetListOfParamEnums(SwaggerParameter swagge
...
 
-        public void SetOptionalValue(ObservableList<AppModelParameter> AppModelParameters, Dictionary<string, string> listExampleValues)
+        public static Dictionary<string, string> ExampleValueDict(SwaggerOperation operations)
+        {
+
+            Dictionary<string, string> exampleValues = new Dictionary<string, string>();
+            
+            try
+            {
+                if (operations.RequestBody.Content.ElementAt(0).Value.Examples != null && operations.RequestBody != null)
```
Comment (Maheshkale447) [inline Ginger/GingerAutoPilot/APIModelLib/SwaggerApi/OpenApiBase.cs:355]: Add operations.RequestBody != null this condition before the first condition

File: Ginger/GingerCoreCommon/GeneralLib/JSONLib/JsonSchemaTools.cs [hunk 5]
```diff
@@ -234,21 +250,39 @@ private static object GenerateJsonObjectFromJsonSchema4(JsonProperty value, List
...
+                        break;
+                    }
+                    output = new JValue(value.Example ?? $"<{value.Type}>");
                     break;
                 case JsonObjectType.Number:
                     output = new JValue(1);
                     break;
                 case JsonObjectType.Integer:
-                    output = new JValue(1);
+                    output = new JValue(value.Example ?? 1);
```
Comment (Maheshkale447) [inline Ginger/GingerCoreCommon/GeneralLib/JSONLib/JsonSchemaTools.cs:264]: please set example value for Number and boolean if present just like it is done for Integer

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

**Comment by Maheshkale447** [comment]
@coderabbitai, please review the PR

File: Ginger/GingerAutoPilot/APIModelLib/SwaggerApi/OpenApiBase.cs (hunk 1 above)
Comment (AmanPrasad43) [reply Ginger/GingerAutoPilot/APIModelLib/SwaggerApi/OpenApiBase.cs:78]: Not in use, removed.

File: Ginger/GingerAutoPilot/APIModelLib/SwaggerApi/OpenApiBase.cs (hunk 2 above)
Comment (AmanPrasad43) [reply Ginger/GingerAutoPilot/APIModelLib/SwaggerApi/OpenApiBase.cs:301]: Setting default only one value.

File: Ginger/GingerAutoPilot/APIModelLib/SwaggerApi/OpenApiBase.cs (hunk 3 above)
Comment (AmanPrasad43) [reply Ginger/GingerAutoPilot/APIModelLib/SwaggerApi/OpenApiBase.cs:307]: removed, it was not required in else if block.

File: Ginger/GingerCoreCommon/GeneralLib/JSONLib/JsonSchemaTools.cs (hunk 5 above)
Comment (AmanPrasad43) [reply Ginger/GingerCoreCommon/GeneralLib/JSONLib/JsonSchemaTools.cs:264]: Added.

File: Ginger/GingerAutoPilot/APIModelLib/SwaggerApi/OpenApiBase.cs (hunk 4 above)
Comment (AmanPrasad43) [reply Ginger/GingerAutoPilot/APIModelLib/SwaggerApi/OpenApiBase.cs:355]: added

File: Ginger/GingerAutoPilot/APIModelLib/SwaggerApi/OpenApiBase.cs [hunk 6]
```diff
@@ -281,23 +284,58 @@ public ObservableList<OptionalValue> GetListOfParamEnums(SwaggerParameter swagge
...
                             ItemName = item.ToString(),
+                            IsDefault = true
                         };
                         lstOptions.Add(value);
                     }
 
                 }
+                else if (swaggerParameter.ActualSchema.ActualProperties.Count>0 || swaggerParameter.ActualSchema.Enumeration.Count>0)
+                {
+                    if(swaggerParameter.ActualSchema.ActualProperties.Count > 0)
```
Comment (Maheshkale447) [inline Ginger/GingerAutoPilot/APIModelLib/SwaggerApi/OpenApiBase.cs:308]: Its still here for ActualProperties

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

**Review by Maheshkale447 — APPROVED** [review]

**Pull request merged**